Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -425,5 +430,42 @@ TEST_F(SerializerTest, IncompleteTag) {
       Status(FirestoreErrorCode::DataLoss, "ignored"), bytes);
 }
 
+TEST_F(SerializerTest, EncodesKey) {
+  EXPECT_EQ("projects/p/databases/d/documents",
+            serializer.EncodeKey(DocumentKey()));
+  EXPECT_EQ(
+      "projects/p/databases/d/documents/one/two/three/four",
+      serializer.EncodeKey(DocumentKey::FromPathString("one/two/three/four")));
+}
+
+TEST_F(SerializerTest, DecodesKey) {
+  EXPECT_EQ(DocumentKey(),
+            serializer.DecodeKey("projects/p/databases/d/documents"));
+  EXPECT_EQ(DocumentKey::FromPathString("one/two/three/four"),
+            serializer.DecodeKey(
+                "projects/p/databases/d/documents/one/two/three/four"));
+  EXPECT_EQ(DocumentKey::FromPathString("one/two/three/four"),
```

Comment:
Consider adding a brief comment, like "With a leading slash", because the two cases look very similar.